src/InvestmentManager.sol

```diff
@@ -419,22 +419,29 @@ contract InvestmentManager is Auth {
 
         // If there's any unclaimed deposits, claim those first
         InvestmentState storage state = investments[liquidityPool][user];
-        if (state.maxMint > 0) {
-            _processDeposit(state, state.maxMint, liquidityPool, user);
+        uint128 tokensToTransfer = trancheTokenAmount;
+        if (state.maxMint >= trancheTokenAmount) {
+            state.maxMint = state.maxMint - trancheTokenAmount;
+            tokensToTransfer = 0;
+        } else if (state.maxMint > 0) {
+            tokensToTransfer = trancheTokenAmount - state.maxMint;
+            state.maxMint = 0;
```

Comment:
Any reason you didn't do it this way? I think the ordering of lines here being consistent is clearer, and the comments are helpful.
```suggestion
        uint128 tokensToTransfer = trancheTokenAmount;
        if (state.maxMint >= trancheTokenAmount) {
            // The full redeem request is covered by the claimable amount
            state.maxMint = state.maxMint - trancheTokenAmount;
            tokensToTransfer = 0;
        } else if (state.maxMint > 0) {
            // The redeem request is only partially covered by the claimable amount
            state.maxMint = 0;
            tokensToTransfer -= state.maxMint;
        }
```

Reply:
I can add the comments, buuut if you can not set ` state.maxMint` to `0 ` and try to deduct the initial value after, it will be 0 already 🙈

Reply:
Oh right. But then we swap the lines in the other case?